docs/manual/resource-leak-checker.tex

```diff
@@ -632,16 +632,18 @@
 then verify the code using a method other than the Resource Leak Checker.
 
 
-%% TODO: Uncomment when the feature is ready to be publicized.
-% \sectionAndLabel{Resource Leak Checker Annotation Inference Algorithm}{resource-leak-checker-inference-algo}
-%
-% We are developing a special inference mechanism to infer annotations for
-% the Resource Leak Checker. By default, this mechanism operates when the
-% -Ainfer flag is enabled. To utilize the whole-program inference mechanism outlined in
-% section \ref{whole-program-inference} in addition to this mechanism for the Resource
-% Leak Checker and its sub-checkers, you should provide the \<-AenableWpiForRlc> flag.
-%
-% The paper \href={https://arxiv.org/pdf/2306.11953.pdf}{Automatic Inference of Resource Leak Specifications}, published at OOPSLA 2023, gives more details on the inference algorithm. Currently, only the first phase of the algorithm described in the paper is available, and the complete implementation will be available soon.
+\sectionAndLabel{Resource Leak Checker Annotation Inference Algorithm}{resource-leak-checker-inference-algo}
+
+We have developed a specialized algorithm to infer annotations for
+the Resource Leak Checker. The RLC uses this algorithm by default
+when the \<-Ainfer> flag is passed, instead of the standard
+whole-program-inference (WPI) algorithm described in Section
+\ref{whole-program-inference}. To use WPI in addition to the specialized algorithm for the Resource
```

Comment:
What are the pros and cons of using WPI in addition to the specialized algorithm?  Why would one want to use/not use this feature?

Also, just passing the flag doesn't cause WPI to be run.  Only the `wpi.sh` script does that.  Maybe you mean that when running that script, the user can (optionally?) pass the `-AenableWpiForRlc` flag?

Reply:
I honestly don't know why a user would run the standard WPI algorithm alongside the specialized algorithm.  @kelloggm any thoughts here?  At this point, I might consider `-AenableWpiForRlc` to be an internal debugging / testing flag that doesn't really need to be documented for users.

I also agree that mentioning inference-related flags here is not super relevant.  I propose we just say (1) when WPI is enabled with the Resource Leak Checker, a specialized algorithm is used, as described in the paper, and (2) you just run `wpi.sh` as described in the later section to run the specialized algorithm, without mentioning the `-Ainfer` flag.  @kelloggm @mernst WDYT?

Reply:
> I honestly don't know why a user would run the standard WPI algorithm alongside the specialized algorithm.

I also can't think of a reason that a typical user would want to do that. We added the flag, as I recall, to enable us to make comparisons between the two modes if we so desired.

> At this point, I might consider -AenableWpiForRlc to be an internal debugging / testing flag that doesn't really need to be documented for users.

I think it is fine to not mention the flag, given my answer above that I can't see why a user would want to use it (except for scientific reasons). From the perspective of a user of the RLC, RLC inference is strictly better than "standard" WPI.

> I propose we just say (1) when WPI is enabled with the Resource Leak Checker, a specialized algorithm is used, as described in the paper, and (2) you just run wpi.sh as described in the later section to run the specialized algorithm, without mentioning the -Ainfer flag

Your proposal sounds good to me.